orm/indexer.go

```diff
@@ -6,52 +6,53 @@ import (
 )
 
 // IndexerFunc creates one or multiple index keys for the source object.
-type IndexerFunc func(value interface{}) ([]RowID, error)
+type IndexerFunc func(value interface{}) ([]interface{}, error)
 
 // IndexerFunc creates exactly one index key for the source object.
-type UniqueIndexerFunc func(value interface{}) (RowID, error)
+type UniqueIndexerFunc func(value interface{}) (interface{}, error)
 
 // Indexer manages the persistence for an Index based on searchable keys and operations.
 type Indexer struct {
-	indexerFunc   IndexerFunc
-	addFunc       func(store sdk.KVStore, codec IndexKeyCodec, secondaryIndexKey []byte, rowID RowID) error
-	indexKeyCodec IndexKeyCodec
+	indexerFunc IndexerFunc
+	addFunc     func(store sdk.KVStore, secondaryIndexKey interface{}, rowID RowID) error
```

Comment:
Is it possible that the index key has multiple parts or are we just allowing one part?

Reply:
good point, we're just allowing one part atm but that should be enforced at the code level, otherwise `stripRowID` will return wrong values in case of multiple parts within the `secondaryIndexKey`